ParraFeedback/Views/Modals/ParraCardModalViewController.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  ParraCardModalViewController.swift
+//  ParraFeedback
+//
+//  Created by Mick MacCallum on 6/18/23.
+//  Copyright © 2023 Parra, Inc. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import ParraCore
+
+internal class ParraCardModalViewController: UIViewController {
+    internal let cardView: ParraCardView
+    internal let modalType: ParraCardModalType
+    internal let transitionStyle: ParraCardModalTransitionStyle
+
+    private var onDismiss: (() -> Void)?
+
+    init(
+        cards: [ParraCore.ParraCardItem],
+        config: ParraCardViewConfig,
+        transitionStyle: ParraCardModalTransitionStyle,
+        modalType: ParraCardModalType,
+        onDismiss: (() -> Void)? = nil
+    ) {
+        self.transitionStyle = transitionStyle
+        self.modalType = modalType
+        self.onDismiss = onDismiss
+
+        // We count on the fact that we control when these modals are presented to know that
+        // if one is instantiated, presentation is inevitable. This allows us to guarentee that
+        // the impression event for the modal is submitted before the card view starts laying out cards.
+        Parra.logAnalyticsEvent(ParraSessionEventType.impression(
```

Comment:
Not priority but a protocol for analytics events could allow for a much cleaner interface at the point of use:
```swift
Parra.log(event: ParraImpressionEvent(modalType: modalTyle))
```

Reply:
All details are hidden here

Reply:
@ianmaccallum totally agree. I was thinking the same yesterday. I'm going to merge this and then start working on merging the modules since there was some weirdness with internal vs external events anyway which will be simplified.